File: src/main/java/toker/warbandscripts/panel/entity/NoticeBoardAccess.java
package toker.warbandscripts.panel.entity;

import com.fasterxml.jackson.annotation.JsonView;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "notice_board_access")
public class NoticeBoardAccess {
    private Integer id;
    private Boolean isOwner = false;
    private NoticeBoard board;
    private Player player;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    @Basic
    @Column(name = "is_owner", nullable = false)
    public Boolean getIsOwner() {
        return isOwner;
    }

    public void setIsOwner(Boolean isOwner) {
        this.isOwner = isOwner;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        NoticeBoardAccess that = (NoticeBoardAccess) o;
        return Objects.equals(id, that.id) &&
                Objects.equals(isOwner, that.isOwner);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, isOwner);
    }

    @ManyToOne
    @JoinColumns({
            @JoinColumn(name = "board_index", referencedColumnName = "index", nullable = false),
            @JoinColumn(name = "board_server_id", referencedColumnName = "server_id", nullable = false)
    })
    @JsonView(View.Board.class)
    public NoticeBoard getBoard() {
        return board;
    }

    public void setBoard(NoticeBoard board) {
        this.board = board;
    }

    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id", nullable = false)
    @JsonView(View.Player.class)
    public Player getPlayer() {
        return player;
    }

    public void setPlayer(Player player) {
        this.player = player;
    }

    public static class View {
        public static class Board {}
        public static class Player {}
    }
}

File: src/main/java/toker/panel/entity/NoticeBoardAccess.kt
package toker.panel.entity

import com.fasterxml.jackson.annotation.JsonView
import toker.panel.entity.NoticeBoardAccess.View.Board
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "notice_board_access")
class NoticeBoardAccess(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(name = "id", nullable = false)
        var id: Int? = null,

        @Column(name = "is_owner", nullable = false)
        var isOwner: Boolean? = false,

        @ManyToOne
        @JoinColumns(JoinColumn(name = "board_index", referencedColumnName = "index", nullable = false), JoinColumn(name = "board_server_id", referencedColumnName = "server_id", nullable = false))
        @JsonView(Board::class)
        var board: NoticeBoard? = null,

        @ManyToOne
        @JoinColumn(name = "user_id", referencedColumnName = "id", nullable = false)
        @JsonView(View.Player::class)
        var player: Player? = null
) {
    interface View {
        interface Board
        interface Player
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as NoticeBoardAccess
        if (id != other.id) return false
        return true
    }

    override fun hashCode(): Int {
        return id ?: 0
    }
}